sklearn/cluster/_dbscan_inner.pyx

```diff
@@ -0,0 +1,48 @@
+# Fast inner loop for DBSCAN.
+# Author: Lars Buitinck
+# License: 3-clause BSD
+
+cimport cython
+from libcpp.vector cimport vector
+cimport numpy as np
```

Comment:
Annoyingly, `cimport numpy` increases the source file size by thousands of lines. Using memoryviews is even worse.